salt/beacons/ps.py

```diff
@@ -31,62 +50,85 @@ def validate(config):
     Validate the beacon configuration
     """
     # Configuration for ps beacon should be a list of dicts
-    if not isinstance(config, list):
-        return False, "Configuration for ps beacon must be a list."
+    if not isinstance(config, dict):
+        return (
+            False,
+            "Configuration for ps beacon must be a dictionary with key 'processes' that contains a list.",
+        )
     else:
-        config = salt.utils.beacons.list_to_dict(config)
-
-        if "processes" not in config:
+        if "processes" not in list(config.keys()):
             return False, "Configuration for ps beacon requires processes."
         else:
-            if not isinstance(config["processes"], dict):
+            if not isinstance(config["processes"], list):
                 return False, "Processes for ps beacon must be a dictionary."
 
+        for e in config["processes"]:
+            status = next(iter(e.values()))
+            if not next(iter(status.values())) in __accepted_statuses__:
+                return (
+                    False,
+                    f"Status not supported, currently supported are {', '.join(__accepted_statuses__)}.",
+                )
+
     return True, "Valid beacon configuration"
 
 
 def beacon(config):
     """
-    Scan for processes and fire events
-
-    Example Config
-
-    .. code-block:: yaml
-
-        beacons:
-          ps:
-            - processes:
-                salt-master: running
-                mysql: stopped
-
-    The config above sets up beacons to check that
-    processes are running or stopped.
+    Search for given process(es) by name and optionally username.
```

Comment:
Add back a proper Example Config, especially if you're not going to document the `config` parameter.